src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -841,6 +841,13 @@ protected override Expression VisitConstant(ConstantExpression node)
                 Out('M');
                 return node;
             }
+            if (Equals(node.Value, default(DateTime)))
```

Comment:
What about the rest?  e,g,
```
DateTimeOffset,
DateOnly,
TimeOnly
```

Also, you can prefilter this check since the constraint has underlying type property.

Reply:
Added them